File: Python/Czy_umiesz_potegowac.py
def last_digit_of_power(base, exponent):
    if exponent == 0:
        return 1
    exponent %= 4
    if exponent == 0:
        exponent = 4
    output = (base ** exponent) % 10
    return output

D = int(input())
for _ in range(D):
    base, exponent = map(int, input().split())
    print(last_digit_of_power(base, exponent))

File: Java/Czy umiesz potegowac.java
import java.util.Scanner;

public class Main 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        int D = input.nextInt();
        for (int i = 0; i < D; i++) 
        {
            int base = input.nextInt();
            int exponent = input.nextInt();
            System.out.println(lastDigitOfPower(base, exponent));
        }
        input.close();
    }

    private static int lastDigitOfPower(int base, int exponent) 
    {
        base = base % 10;
        if (exponent == 0)
            return 1;
        int expMod = exponent % 4;
        if (expMod == 0)
            expMod = 4;
        int result = 1;
        for (int i = 0; i < expMod; i++)
            result = (result * base) % 10;
        return result;
    }
}
